x-pack/legacy/plugins/apm/public/hooks/useTransactionBreakdown.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// import { useMemo } from 'react';
+import { IUrlParams } from '../context/UrlParamsContext/types';
+import { useUiFilters } from '../context/UrlParamsContext';
+import { useFetcher } from './useFetcher';
+import { callApi } from '../services/rest/callApi';
+import { getUiFiltersES } from '../services/ui_filters/get_ui_filters_es';
+import { TransactionBreakdownAPIResponse } from '../../server/lib/transactions/breakdown';
+
+export function useTransactionBreakdown(urlParams: IUrlParams) {
+  const { serviceName, start, end, transactionName } = urlParams;
```

Comment:
We may want to call the `useUrlParams` hook within `useTransactionBreakdown`:
```
const { urlParams, uiFilters } = useUrlParams();
```

Reply:
aye, changed, removes some unnecessary prop drilling 👍